tests/main/assumes/task.yaml

```
@@ -0,0 +1,44 @@
+summary: Test that assumes work correctly
+
+execute: |
+    echo "Try to install an impossible version"
+    if snap install --edge test-snapd-assumes 2>stderr.txt; then
+        echo "snap install should not work here"
+        exit 1
+    fi
+    echo "Ensure we got the right error"
+    MATCH "unsupported feature: snapd9.99" < stderr.txt
+    echo "Ensure we got the error early"
+    not grep "Mount snap" < stderr.txt
```

FYI, you should be able to use `not` with `MATCH` correctly.
```suggestion
    not MATCH "Mount snap" < stderr.txt
```

---

Thanks, however I think we want grep here. Lets assume I have "Mount snap" in stderr. With MATCH it would suppress the grep output (because the MATCH matched). But in this case I'm actually interested in the details, i.e. the stderr.txt line may contain something interesting. So grep seems to be more appropriate here. Or am I missing something?

---

Ah, good point. Sorry for my knee-jerk reaction.